js/matchingModel.js

```diff
@@ -0,0 +1,227 @@
+
+            this.get('_items').forEach(function(item) {
+                item._options = _.shuffle(item._options);
+            });
+        },
+
+        restoreUserAnswers: function() {
+            if (!this.get('_isSubmitted')) return;
+
+            var userAnswer = this.get('_userAnswer');
+
+            this.get('_items').forEach(function(item, index) {
+                item._options.forEach(function(option, index) {
+                    if (option._index == userAnswer[item._index]) {
+                        option._isSelected = true;
+                        item._selected = option;
+                    }
+                });
+            });
+
+            this.setQuestionAsSubmitted();
+            this.markQuestion();
+            this.setScore();
+            this.setupFeedback();
+        },
+
+        canSubmit: function() {
+            // can submit if every item has a selection
+            var canSubmit = _.every(this.get('_items'), function(item) {
+                return _.findWhere(item._options, {'_isSelected':true}) !== undefined;
+            });
+
+            return canSubmit;
+        },
+
+        setOptionSelected: function(itemIndex, optionIndex, isSelected) {
+            var item = this.get('_items')[itemIndex];
+            var option = item._options[optionIndex];
+
+            option._isSelected = isSelected;
+            item._selected = option;
+        },
+
+        storeUserAnswer: function() {
+
+            var userAnswer = new Array(this.get('_items').length);
+            var tempUserAnswer = new Array(this.get('_items').length);
+
+            this.get('_items').forEach(function(item, index) {                
+                var optionIndex = _.findIndex(item._options, function(o) {return o._isSelected});
+
+                tempUserAnswer[item._index] = optionIndex;
+                userAnswer[item._index] = item._options[optionIndex]._index;
+            }, this);
+
+            this.set({
+                '_userAnswer': userAnswer,
+                '_tempUserAnswer': tempUserAnswer
+            });
+        },
+
+        isCorrect: function() {
+            var numberOfCorrectAnswers = 0;
+
+            this.get('_items').forEach(function(item, index) {
+
+                var isCorrect = (item._selected && item._selected._isCorrect);
+
+                if (!isCorrect) {
+                    item._isCorrect = false;
+                    return;
+                }
+
+                numberOfCorrectAnswers++;
+                item._isCorrect = true;
+                this.set({
+                    '_numberOfCorrectAnswers': numberOfCorrectAnswers,
+                    '_isAtLeastOneCorrectSelection': true
+                });
+
+            }, this);
+
+            this.set('_numberOfCorrectAnswers', numberOfCorrectAnswers);
+
+            if (numberOfCorrectAnswers === this.get('_items').length) {
+                return true;
+            }
+
+            return false;
```

Comment:
these five lines could be reduced to `return numberOfCorrectAnswers === this.get('_items').length`